### PR TITLE
Fix building as dependency with add_subdirectory()

### PR DESCRIPTION
**Pull request by JoelLinn**
- When building as a dependency using cmake's add_subdirectory() use the
  current source directory instead of the top level source tree.

Previously it was not possible to use libphonenumber in such a way, especially when linking statically.

### CODE REVIEW

**Comment by JoelLinn** [comment]
@googlebot I signed it!

**Comment by JoelLinn** [comment]
Rebased.

**Comment by JoelLinn** [comment]
I would be happy if anybody (maintainer or not) would be able to at least comment on this.

**Comment by penmetsaa** [comment]
Thanks much for the change. Sorry for the delay here; because of some setup issues at my end, I could not test / look into it.
This change looks and sounds innocuous; and beneficiary w.r.t buidling from the end products points of view.
Followed regular CPP build process with this change and all tests passed.

**Review by penmetsaa — DISMISSED** [review]

**Comment by penmetsaa** [comment]
Please make the branch sync with master. Will merge after it.

**Review by gvollant — DISMISSED** [review]

**Review by penmetsaa — DISMISSED** [review]

**Comment by penmetsaa** [comment]
Please sync the branch with master.
Out of date because of new merges.

**Review by penmetsaa — APPROVED** [review]

**Pull request merged**
